bike_kollective/lib/ui/explore_bikes/explore_bikes_screen.dart

```diff
@@ -19,13 +24,60 @@ class ExploreBikesScreenState extends ConsumerState<ExploreBikesScreen> {
 
   @override
   Widget build(BuildContext context) {
-    var bikes = ref.watch(availableBikesProvider);
+    final bikes = ref.watch(availableBikesProvider);
+
     if (bikes.isEmpty) {
       return const Center(child: Text('No bikes available'));
     }
 
-    return Center(
-      child: BikesViewer(availableBikes: bikes, isMyBikes: false),
+    return FutureBuilder<BKGeoPoint>(
+      future: ref.read(userLocationProvider).getCurrent(),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(child: CircularProgressIndicator());
+        }
+
+        if (snapshot.hasError || !snapshot.hasData) {
+          return const Center(child: Text('Error retrieving location'));
+        }
+
+        final userLocation = snapshot.data!;
+
+        final bikesWithDistances = bikes.map((bike) {
+          final distance = _calculateDistance(userLocation, bike.locationPoint);
+          return bike.copyWith(distance: distance);
+        }).toList()
+          ..sort((a, b) => a.distance!.compareTo(b.distance!));
+
+        return Center(
+          child: BikesViewer(
+            availableBikes: bikesWithDistances,
+            isMyBikes: false,
+          ),
+        );
+      },
     );
   }
+
+  double _calculateDistance(BKGeoPoint userLocation, BKGeoPoint bikeLocation) {
```

Comment:
Consider moving this to its own file, in a utils/ directory. Then we can call this where we need the distance, it's ok to run this multiple times